components/console/helpers/progressindicator.rst

```diff
@@ -0,0 +1,126 @@
+Progress Indicator
+==================
+
+When executing longer-running commands without knowing if the the processing
+is nearly done or not, it may be helpful to show that something is actually
+happening and that updates as your command runs.
+
+To do so, use the
+:class:`Symfony\\Component\\Console\\Helper\\ProgressIndicator` and advance the
+progress as the command executes::
+
+    use Symfony\Component\Console\Helper\ProgressIndicator;
+
+    // creates a new progress indicator
+    $progressIndicator = new ProgressIndicator($output);
+
+    // starts and displays the progress indicator with a custom message
+    $progressIndicator->start('Processing...');
+
+    $i = 0;
+    while ($i++ < 50) {
+        // ... do some work
+
+        // advances the progress indicator
+        $progressIndicator->advance();
+    }
+
+    // ensures that the progress indicator shows a final message
+    $progressIndicator->finish('Finished');
+
+Customizing the Progress Indicator
+----------------------------------
+
+Built-in Formats
+~~~~~~~~~~~~~~~~
+
+By default, the information rendered on a progress indicator depends on the current
+level of verbosity of the ``OutputInterface`` instance:
+
+.. code-block:: text
+
+    # OutputInterface::VERBOSITY_NORMAL (CLI with no verbosity flag)
+     \ Processing...
+     | Processing...
+     / Processing...
+     - Processing...
+
+    # OutputInterface::VERBOSITY_VERBOSE (-v)
+     \ Processing... (1 sec)
+     | Processing... (1 sec)
+     / Processing... (1 sec)
+     - Processing... (1 sec)
+
+    # OutputInterface::VERBOSITY_VERY_VERBOSE (-vv) and OutputInterface::VERBOSITY_DEBUG (-vvv)
+     \ Processing... (1 sec, 6.0 MiB)
+     | Processing... (1 sec, 6.0 MiB)
+     / Processing... (1 sec, 6.0 MiB)
+     - Processing... (1 sec, 6.0 MiB)
+
+.. note::
+
+    If you call a command with the quiet flag (``-q``), the progress indicator won't
+    be displayed.
+
+Instead of relying on the verbosity mode of the current command, you can also
+force a format via the second argument of the ``ProgressIndicator``
+constructor::
+
+    $progressIndicator = new ProgressIndicator($output, 'verbose');
+
+The built-in formats are the following:
+
+* ``normal``
+* ``verbose``
+* ``very_verbose``
+
+If your terminal doesn't support ANSI, use the ``no_ansi`` variants:
+
+* ``normal_no_ansi``
+* ``verbose_no_ansi``
+* ``very_verbose_no_ansi``
+
+Custom Indicator Values
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Instead of using the built-in indicator values, you can also set your own::
+
+    $progressIndicator = new ProgressIndicator($output, 'verbose', 100, ['o', 'O']);
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**🎯 suggestion:** ‏Could we use `['⠏', '⠛', '⠹', '⢸', '⣰', '⣤', '⣆', '⡇']` to show a more "modern" spinner?

Reply:
Oh yeah definitely, thanks!